chapter05/Task_5_2.py

```diff
@@ -0,0 +1,117 @@
+# Task 5.2
+# Program about hero and his specifications
+#  User can customize hero
+
+specification = {
+    'Strength': 0,
+    'health': 0,
+    'wisdom': 0,
+    'agility': 0}
+
+points = 30
+
+print('''
+        Hello and Welcome
+    Today you customize your hero
+
+    Let's start:''')
+
+while True:
+    choice = input('''
+1. Start to customize hero
+2. Edit strength
+3. Edit health
+4. Edit wisdom
+5. Edit agility
+
+Remember, you can only distribute {} points\n'''.format(points))
+
+    if choice == '1':
+        for item in specification:
+            while points >= 0:
+                how_many = int(input("How many points do you want to bet " + item.lower() + '?\n'))
+                if points + how_many < 0 or points - int(how_many) >30:
+                    print("You can't do this. you don't have enough points")
+                    print('You have left', points, 'points.')
+                elif int(specification[item]) + int(points) + int(how_many) < 0:
+                    print('Enter only positive values')
+                    print('You have left', points, 'points.')
+                else:
+                    points = points - how_many
+                    specification[item] = int(specification[item]) + int(how_many)
+                    break
+            print('You have left', points, 'points.')
+
+    elif choice == '2':
```

Comment:
Here and next the core is just a duplication of the same thing, only the specification is different, so I suggest you to reformat is the following:
```
else:
   item = {
       "1": "strength",
       "2": "health",
       "3": "wisdom",
       "4": "agility",
   }.get(choice)
   if item is None:
       print("Menu hasn't this number")
       continue
   # All the code for a single specification, without any repeats you've had.
```

Reply:
```
items = {
       "0": specification.keys(),
       "1": ["strength"],
       "2": ["health"],
       "3": ["wisdom"],
       "4": ["agility"],
   }.get(choice)
   if item is None:
       print("Menu hasn't this number")
       continue
   for item in items:
        # All the code for a single specification, without any repeats you've had.
   
```